################################################################################
SHORT DESCRIPTION:
################################################################################
Fix libxl memory corruption due to incorrect array sizing.

################################################################################
LONG DESCRIPTION:
################################################################################

Bug introduced upstream in: 4c8153d97efe087f139a28c096f825e8069a3931
# Date:   Tue Mar 28 09:42:26 2017 +0100

Fixed upstream in: 614a14736e33fb84872eb00f08799ebbc73a96c6
# Date:   Thu Jul 13 10:03:39 2017 +0800

but not yet present in 4.9 release.

################################################################################
CHANGELOG
################################################################################
introduced for Xen 4.9

################################################################################
REMOVAL
################################################################################
This fix has been merged into the upstream stable-4.9 branch and will be
included in the next Xen release, at which point this patch can be dropped.

################################################################################
UPSTREAM PLAN
################################################################################
Already there.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
None

################################################################################
PATCHES
################################################################################

Index: xen-4.9.0/tools/libxl/libxl_create.c
===================================================================
--- xen-4.9.0.orig/tools/libxl/libxl_create.c
+++ xen-4.9.0/tools/libxl/libxl_create.c
@@ -475,7 +475,7 @@ int libxl__domain_build(libxl__gc *gc,
         vments[4] = "start_time";
         vments[5] = GCSPRINTF("%lu.%02d", start_time.tv_sec,(int)start_time.tv_usec/10000);
 
-        localents = libxl__calloc(gc, 9, sizeof(char *));
+        localents = libxl__calloc(gc, 11, sizeof(char *));
         i = 0;
         localents[i++] = "platform/acpi";
         localents[i++] = libxl__acpi_defbool_val(info) ? "1" : "0";
